### PR TITLE
Document allowEmptyInput option for the node js API

### LINKED ISSUE #5866: Document allowEmptyInput for Node API
### Describe the documentation issue

Does the Node.js API have an equivalent for the `--allow-empty-input`  CLI argument?
We're working on a project where we're slowly updating an old codebase and most of the styles are ignored by default. 

### What solution would you like to see?

A way to allow empty input via the Node API in case all passed files are ignored in `.stylelintignore`

### PR DESCRIPTION
**Pull request by doing-art**
> Which issue, if any, is this issue related to?

Closes #5866 

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self-explanatory."

### CODE REVIEW

**Review by jeddy3 — APPROVED** [review]
LGTM, thanks!

**Review by ybiquitous — COMMENTED** [review]
[ask] This change sounds good to me. But in addition, is it better to update `usage/options.md` and `usage/cli.md`?

- `usage/options.md` describes the shared options between CLI and Node.js API
- `usage/cli.md` has "More info" links to `usage/options.md`

**Comment by jeddy3** [comment]
Sounds good. Let's move the option from the Node.API page to the options one, and add a `more info` link to the cli page (https://stylelint.io/user-guide/usage/cli#--allow-empty-input---aei) to be consistent.

**Comment by doing-art** [comment]
> [ask] This change sounds good to me. But in addition, is it better to update `usage/options.md` and `usage/cli.md`?
> 
>     * `usage/options.md` describes the shared options between CLI and Node.js API
> 
>     * `usage/cli.md` has "More info" links to `usage/options.md`

Thank you for

**Comment by doing-art** [comment]
@jeddy3 @ybiquitous Thank you for your comments. Updated the PR.

**Review by jeddy3 — APPROVED** [review]
Thanks!

File: docs/user-guide/usage/cli.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ The CLI accepts:
 
 ### `--allow-empty-input, --aei`
 
-The process exits without throwing an error when glob pattern matches no files.
+The process exits without throwing an error when glob pattern matches no files. [More info](options.md#allowEmptyInput)
```
Comment (ybiquitous) [inline docs/user-guide/usage/cli.md:17]: [trivial] Missing period?

```suggestion
The process exits without throwing an error when glob pattern matches no files. [More info](options.md#allowEmptyInput).
```

**Review by ybiquitous — APPROVED** [review]
Thank you!

**Review by ybiquitous — APPROVED** [review]

**Pull request merged**